--- a/libgo/configure.ac
+++ b/libgo/configure.ac
@@ -549,10 +549,10 @@
 [libgo_cv_lib_pthread=yes],
 [libgo_cv_lib_pthread=no])
 CFLAGS=$CFLAGS_hold])
-PTHREAD_CFLAGS=
+PTHREAD_CFLAGS="-L$(STAGING_DIR_ROOT)/lib"
 if test "$libgo_cv_lib_pthread" = yes; then
   # RISC-V apparently adds -latomic when using -pthread.
-  PTHREAD_CFLAGS="-pthread -L../libatomic/.libs"
+  PTHREAD_CFLAGS="-L$(STAGING_DIR_ROOT)/lib -pthread -L../libatomic/.libs"
 fi
 AC_SUBST(PTHREAD_CFLAGS)
 
--- a/libgo/configure
+++ b/libgo/configure
@@ -14701,10 +14701,10 @@
 fi
 { $as_echo "$as_me:${as_lineno-$LINENO}: result: $libgo_cv_lib_pthread" >&5
 $as_echo "$libgo_cv_lib_pthread" >&6; }
-PTHREAD_CFLAGS=
+PTHREAD_CFLAGS="-L$(STAGING_DIR_ROOT)/lib"
 if test "$libgo_cv_lib_pthread" = yes; then
   # RISC-V apparently adds -latomic when using -pthread.
-  PTHREAD_CFLAGS="-pthread -L../libatomic/.libs"
+  PTHREAD_CFLAGS="-L$(STAGING_DIR_ROOT)/lib -pthread -L../libatomic/.libs"
 fi
 
 
--- a/gotools/Makefile.in
+++ b/gotools/Makefile.in
@@ -252,7 +252,7 @@
 INSTALL_PROGRAM = @INSTALL_PROGRAM@
 INSTALL_SCRIPT = @INSTALL_SCRIPT@
 INSTALL_STRIP_PROGRAM = @INSTALL_STRIP_PROGRAM@
-LDFLAGS = @LDFLAGS@
+LDFLAGS = -L$(STAGING_DIR_ROOT)/lib @LDFLAGS@
 LIBOBJS = @LIBOBJS@
 LIBS = @LIBS@
 LTLIBOBJS = @LTLIBOBJS@
@@ -348,7 +348,7 @@
 GOCOMPILE = $(GOCOMPILER) $(GOCFLAGS)
 AM_GOCFLAGS = -I $(libgodir)
 AM_LDFLAGS = -L $(libgodir) -L $(libgodir)/.libs \
-	-L $(libgccdir) -L $(libgccdir)/.libs -lgcc_s
+	-L $(libgccdir) -L $(libgccdir)/.libs -L$(STAGING_DIR_ROOT)/lib -lgcc_s
 GOLINK = $(GOCOMPILER) $(GOCFLAGS) $(AM_GOCFLAGS) $(LDFLAGS) $(AM_LDFLAGS) -o $@
 libgosrcdir = $(srcdir)/../libgo/go
 cmdsrcdir = $(libgosrcdir)/cmd
